### ISSUE TITLE
configurable buildconfig output

### ISSUE THREAD

**Issue by tbugfinder** [opendevstack/ods-provisioning-app#275]
**Is your feature request related to a problem? Please describe.**
It would be beneficial for hybrid workloads to use OpenShift/ODS as the build environment pushing images to an (or even multiple) external docker registry (registries) like Nexus, AWS ECR or Azure Container Registry.
I'm using the term hybrid workloads in a sense were a single container image is used within OpenShift, AWS, Azure, HPC-Singularity executor or even plain docker.

**Describe the solution you'd like**
The output target should be configurable so that images could be pushed to a docker registry. I didn't research if output could even be a multi-target item (ImageStream *and* docker registries).

**Describe alternatives you've considered**
Building images within ODS and use the same repo for external builds which isn't a 1:1 move as Jenkinsfile has to be changed.
Not everybody wants to expose the internal registry.

**Additional context**
https://docs.openshift.com/container-platform/3.11/dev_guide/builds/build_output.html
https://docs.openshift.com/container-platform/4.2/builds/managing-build-output.html

**Comment by michaelsauter** [comment]
@tbugfinder Makes sense to me. To my understanding though, the options in OpenShift are very limited - you can't even tag the produced image with two tags (I researched this).

So maybe the image would need to be pulled in Jenkins after building, and then pushed to another registry? Would that work for you?

**Comment by tbugfinder** [comment]
@michaelsauter Pulling and pushing images in Jenkins is an option. Would I have to use a custom jenkins_slave to add docker/buildah executable?

**Comment by michaelsauter** [comment]
@michaelsauter Probably. I think pulling might work via `oc`, put pushing would require some different tool.

I am moving this issue to `ods-core` for now until we now where exactly to put it.
